File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        List<Integer> arr = new ArrayList<>();
        while (head != null){
            arr.add(head.val);
            head = head.next;
            }
        
        int idx = 0, res = 0; 
        for (int i = arr.size() - 1; i >= 0; i--){
            res += (arr.get(i) * Math.pow(2,idx));
            idx++;
        }

        return res;
    }
}

File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        arr = []
        while head != None:
            arr.append(head.val)
            head = head.next
            
        idx = res = 0
        for i in range(len(arr) - 1, -1, -1):
            res += (arr[i] * (2 ** idx))
            idx += 1
        
        return res
        